fastvideo/v1/pipelines/implementations/hunyuan/hunyuan_pipeline.py

```diff
@@ -0,0 +1,224 @@
+        
+        self.add_stage(
+            stage_name="input_validation_stage",
+            stage=InputValidationStage()
+        )
+        
+        self.add_stage(
+            stage_name="prompt_encoding_stage_primary",
+            stage=PromptEncodingStage(
+                text_encoder=self.get_module("text_encoder"),
+                is_secondary=False
+            )
+        )
+        
+        self.add_stage(
+            stage_name="prompt_encoding_stage_secondary",
+            stage=PromptEncodingStage(
+                text_encoder=self.get_module("text_encoder_2"),
+                is_secondary=True
+            )
+        )
+        
+        self.add_stage(
+            stage_name="conditioning_stage",
+            stage=ConditioningStage()
+        )
+        
+        self.add_stage(
+            stage_name="timestep_preparation_stage",
+            stage=TimestepPreparationStage(
+                scheduler=self.get_module("scheduler")
+            )
+        )
+        
+        self.add_stage(
+            stage_name="latent_preparation_stage",
+            stage=LatentPreparationStage(
+                scheduler=self.get_module("scheduler")
+            )
+        )
+        
+        self.add_stage(
+            stage_name="denoising_stage",
+            stage=DenoisingStage(
+                transformer=self.get_module("transformer"),
+                scheduler=self.get_module("scheduler")
+            )
+        )
+        
+        self.add_stage(
+            stage_name="decoding_stage",
+            stage=DecodingStage(
+                vae=self.get_module("vae")
+            )
+        )
+    
+    def initialize_pipeline(self, inference_args: InferenceArgs):
+        """
+        Initialize the pipeline.
+        """
+        vae_scale_factor = 2**(len(self.get_module("vae").block_out_channels) - 1)
+        inference_args.vae_scale_factor = vae_scale_factor
+
+        self.image_processor = VaeImageProcessor(vae_scale_factor=vae_scale_factor)
+        self.add_module("image_processor", self.image_processor)
+
+
+        num_channels_latents = self.get_module("transformer").in_channels
+        inference_args.num_channels_latents = num_channels_latents
+
+
+    # TODO
+    def adjust_video_length(self, batch: ForwardBatch, inference_args: InferenceArgs):
+        """Adjust video length based on VAE version"""
+        video_length = batch.num_frames
+        batch.num_frames = (video_length - 1) // 4 + 1
+        return batch
+
+    
+    @torch.no_grad()
+    def forward(self, batch: ForwardBatch, inference_args: InferenceArgs) -> ForwardBatch:
+        logger.info(f"Running pipeline stages: {self._stage_name_mapping.keys()}")
+        logger.info(f"Batch: {batch}")
+        # for stage in self._stages:
+            # batch = stage(batch, inference_args)
+
+        # or 
+
+        batch = self.input_validation_stage(batch, inference_args)
```

Comment:
agree with @comaniac. I am confused here -- in the above you already have clearly defined stages? so the block block below could just be discarded?